### PR TITLE
qsctl command presign modified

### PR DESCRIPTION
**Pull request by Fiile**

### CODE REVIEW

File: bin/qsctl [hunk 1]
```diff
@@ -20,8 +20,10 @@ import sys
 
 import qingstor.qsctl.driver
 
+
```
Comment (Xuanwo) [inline bin/qsctl]: Why add this emppty line ？

File: qingstor/qsctl/commands/presign.py [hunk 2]
```diff
@@ -56,6 +56,15 @@ def generate_presign_url(cls):
             sys.exit(-1)
         cls.validate_bucket(bucket)
         current_bucket = cls.client.Bucket(bucket, cls.bucket_map[bucket])
+
+        current_acl = current_bucket.get_acl()
+        for v in current_acl["acl"]:
+            # go through the whole list
+            if "QS_ALL_USERS" == v["grantee"]["name"] and "FULL_CONTROL" == v["permission"]:
+                obj_url = "https://" + cls.bucket_map[bucket] + ".qingstor.com/" + bucket + "/" + prefix
```
Comment (Xuanwo) [inline qingstor/qsctl/commands/presign.py]: 1. Do not hard code `https` and `qingstor.com` here.

1. Use the following code will be better for understanding:

    ```python
    obj_url = "{protocol}://{bucket_name}.{zone}.{host}/{object_key}".format(
        protocol="https",
        bucket_name="mybucket",
        zone="pek3a",
        host="qingstor.com",
        object_key="test_object"
    )
    ```
1. `obj_url` is not a good name, wo should find a better one

**Comment by Xuanwo** [comment]
And docs should also be updated.

**Comment by Xuanwo** [comment]
Commit message is not so good, please follow these:

- Capitalize the subject line
- Limit the subject line to 50 characters
- Do not end the subject line with a period
- Wrap the body at 72 characters
- Use the body to explain what happened

File: bin/qsctl (hunk 1 above)
Comment (Fiile) [inline bin/qsctl]: I thought it had been added automatically.

File: qingstor/qsctl/commands/presign.py (hunk 2 above)
Comment (Fiile) [inline qingstor/qsctl/commands/presign.py]: ACK

File: qingstor/qsctl/commands/presign.py [hunk 3]
```diff
@@ -56,26 +57,41 @@ def generate_presign_url(cls):
...
-                )
-                uni_print(public_url)
-                return public_url
-
-        prepared = current_bucket.get_object_request(prefix).sign_query(
-            get_current_time() + cls.options.expire_seconds
-        )
-        uni_print(prepared.url)
-        return prepared.url
+        resp = current_bucket.head()
```
Comment (Xuanwo) [inline qingstor/qsctl/commands/presign.py]: Should `head_object(prefix)` here.

File: scenarios/steps/presign.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        "-e",
+    ).stdout.decode("utf-8")
+
+
+@then(u'the link should include parameters like signature, etc.')
+def step_impl(context):
+    result = urllib.parse.urlparse(context.output)
+    params = urllib.parse.parse_qs(result.query, True)
+    object_key = result.path
+    assert_that("signature" in params.keys() and
```
Comment (Xuanwo) [inline scenarios/steps/presign.py]: `assert_that(params).contains("signature")` maybe better, you can read the assertpy's doc <https://github.com/ActivisionGameScience/assertpy> for more infomation.

**Pull request merged**